subsys/bluetooth/audio/bap_unicast_client.c

```diff
@@ -3316,48 +3411,117 @@ static uint8_t unicast_client_pacs_context_read_func(struct bt_conn *conn, uint8
 	return BT_GATT_ITER_STOP;
 }
 
+static uint8_t unicast_client_pacs_context_discover_cb(struct bt_conn *conn,
+						       const struct bt_gatt_attr *attr,
+						       struct bt_gatt_discover_params *discover)
+{
+	struct bt_bap_unicast_client_discover_params *params;
+	struct bt_gatt_chrc *chrc;
+	int err;
+
+	params = CONTAINER_OF(discover, struct bt_bap_unicast_client_discover_params,
+			      discover);
+
+	if (attr == NULL) {
+		LOG_ERR("Unable to find %s PAC context",
+			bt_audio_dir_str(params->dir));
+
+		params->err = BT_ATT_ERR_ATTRIBUTE_NOT_FOUND;
+
+		__ASSERT(params->func != NULL, "params->func was NULL");
+
+		params->func(conn, NULL, NULL, params);
+
+		return BT_GATT_ITER_STOP;
+	}
+
+	chrc = attr->user_data;
+
+	LOG_DBG("conn %p attr %p handle 0x%04x dir %s",
+		conn, attr, chrc->value_handle, bt_audio_dir_str(params->dir));
+
+	/* TODO: Subscribe to PAC context */
+
+	params->read.func = unicast_client_pacs_context_read_func;
+	params->read.handle_count = 1U;
+	params->read.single.handle = chrc->value_handle;
+	params->read.single.offset = 0U;
+
+	err = bt_gatt_read(conn, &params->read);
+	if (err != 0) {
+		LOG_DBG("Failed to read PAC records: %d", err);
+
+		params->err = err;
+
+		params->func(conn, NULL, NULL, params);
+	}
+
+	return BT_GATT_ITER_STOP;
```

Comment:
(as above) if there is more to discover, should it not return BT_GATT_ITER_CONTINUE? - or will there always only be 1?

Reply:
There will only ever be a single PACS context characteristic. 

What this is doing is that it will do a discovery, then a read, then a new discovery, then a new read, etc.